Include version for glance api calls

albertomurillo opened the pull request:
- Add v2 to glance url in NewImageServiceV2

Signed-off-by: Alberto Murillo alberto.murillo.silva@intel.com

albertomurillo commented:
Please do not merge yet. there is bug

sameo commented on openstack/client.go:
```diff
@@ -278,9 +278,7 @@ func NewBlockStorageV1(client *gophercloud.ProviderClient, eo gophercloud.Endpoi
-	return &gophercloud.ServiceClient{ProviderClient: client,
-		Endpoint:     url,
-		ResourceBase: url + "v2/"}, nil
+	return &gophercloud.ServiceClient{ProviderClient: client, Endpoint: url}, nil
```
That one makes perfect sense.

sameo commented on openstack/client.go:
```diff
@@ -321,5 +319,7 @@ func NewImageServiceV2(client *gophercloud.ProviderClient, eo gophercloud.Endpoi
-	return &gophercloud.ServiceClient{ProviderClient: client, Endpoint: url}, nil
+	return &gophercloud.ServiceClient{ProviderClient: client,
+		Endpoint:     url,
+		ResourceBase: url + "v2/"}, nil
```
This one is not needed if keystone's glance endpoints are properly created, i.e. they should include "v2/".

albertomurillo replied:
The thing is, the endpoint is usually not created with "v2/" included.
Since gophercloud is intended to work with any openstack deployment it needs to be compatible with all the openstack deployments out there, otherwise, you need to make all of them update their endpoints

http://docs.openstack.org/mitaka/install-guide-ubuntu/glance-install.html

Commit:
Include version for glance api calls

- Add v2 to glance url in NewImageServiceV2

Signed-off-by: Alberto Murillo <alberto.murillo.silva@intel.com>

Commit:
Dont hide error in Extract functions

The following functions where returning a
decoded result and the error of the decoding
operation without checking if the original
result contains an error already

  func (c CreateResult) Extract() (*Image, error)
  func (c GetResult) Extract() (*Image, error)
  func (u UpdateResult) Extract() (*Image, error)

Signed-off-by: Alberto Murillo <alberto.murillo.silva@intel.com>

Commit:
Fix location for image properties

- Custom image propertoes goes in the body itself, not
  inside a 'properties' field

Signed-off-by: Alberto Murillo <alberto.murillo.silva@intel.com>
